app/controllers/api/v1/users_controller.rb

```diff
@@ -55,22 +60,22 @@ def create
           create_default_room(user)
 
           render_data data: current_user, serializer: CurrentUserSerializer, status: :created
-        elsif user.errors.to_a == ['Email has already been taken']
-          render_error errors: 'EmailAlreadyExists', status: :bad_request
+        elsif user.errors.size == 1 && user.errors.of_kind?(:email, :taken)
+          render_error errors: Rails.configuration.custom_error_msgs[:email_exists], status: :bad_request
         else
-          render_error errors: user.errors.to_a, status: :bad_request
+          render_error errors: Rails.configuration.custom_error_msgs[:record_invalid], status: :bad_request
         end
       end
+      # rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
 
-      # TODO: Add a before_action callback to update,destory and purge_avatar calling a find_user.
       def update
         user = User.find(params[:id])
 
         if user.update(user_params)
           create_default_room(user)
           render_data  status: :ok
         else
-          render_error errors: user.errors.to_a
+          render_error errors: Rails.configuration.custom_error_msgs[:record_invalid]
```

Comment:
Sending the plain active record errors is a bad practice,
We can log them once tackling logging.
However, we need to send an error message without details to the client app that it can leverage later on for feedbacking,